docs/design/services/dynamodb/high-level-library/prototype/option-1/sync/Prototype.java

```diff
@@ -0,0 +1,1026 @@
+     *                                                                .readCapacity(5)
+     *                                                                .writeCapacity(5)
+     *                                                                .build();
+     *
+     *         KeySchema indexKeys = KeySchema.builder()
+     *                                        .putKey("extra-partition-key", ItemAttributeIndexType.PARTITION_KEY)
+     *                                        .build();
+     *
+     *         Table table = client.getTable("my-table");
+     *
+     *         table.createGlobalSecondaryIndex(CreateGlobalSecondaryIndexRequest.builder()
+     *                                                                           .indexName("my-new-index")
+     *                                                                           .provisionedCapacity(tableCapacity)
+     *                                                                           .keySchema(tableKeys)
+     *                                                                           .build());
+     *     }
+     * </code>
+     */
+    CreateGlobalSecondaryIndexResponse createGlobalSecondaryIndex(CreateGlobalSecondaryIndexRequest createRequest);
+
+    /**
+     * Invoke DynamoDB to create or override an {@link Item} in this table.
+     *
+     * This method is optimized for performance, and provides no additional response data. For additional options
+     * like conditions or consumed capacity, see {@link #putItem(PutItemRequest)}.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         Table table = client.getTable("my-table");
+     *         table.putItem(Item.builder()
+     *                           .putAttribute("partition-key", UUID.randomUUID())
+     *                           .putAttribute("creation-time", Instant.now())
+     *                           .build());
+     *     }
+     * </code>
+     */
+    void putItem(Item item);
+
+    /**
+     * Invoke DynamoDB to create or override an {@link Item} in this table.
+     *
+     * This method provides more options than {@link #putItem(Item)}, like conditions or consumed capacity.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         Table table = client.getTable("my-table");
+     *         table.putItem(PutItemRequest.builder()
+     *                                     .item(Item.builder()
+     *                                               .putAttribute("partition-key", "key")
+     *                                               .putAttribute("version", 2)
+     *                                               .build())
+     *                                     .condition("version = :expected_version")
+     *                                     .putConditionAttribute(":expected_version", 1)
```

Comment:
If I'm reading this correctly, it's not super-intuitive to me. Can we link these together better than just strings?

Reply:
This is currently just a passthrough for the Dynamo DB conditions. There will hopefully also be a higher-level abstraction for conditions. I didn't want to design that in the prototype.